plugins/sql_config_validator.h

```diff
@@ -98,6 +111,14 @@ template<typename FieldType> struct NoGroupByHaving
     }
 };
 
+template<typename FieldType> struct NoGroupByHaving<Optional<FieldType> >
```

Comment:
As mentioned, this is getting too complex for no reason.  I plan to revisit it once we I've some bandwith.